internal/pkg/skuba/kured/kured.go

```diff
@@ -18,20 +18,20 @@
 package kured
 
 import (
-	"k8s.io/kubernetes/cmd/kubeadm/app/images"
-
 	"github.com/SUSE/skuba/internal/pkg/skuba/kubernetes"
-	"github.com/SUSE/skuba/pkg/skuba"
 	"github.com/pkg/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
 )
 
-func GetKuredImage() string {
-	return images.GetGenericImage(skuba.ImageRepository, "kured",
-		kubernetes.CurrentAddonVersion(kubernetes.Kured))
-}
+const (
+	kuredDSName             = "kured"
+	kuredLockAnnotationJson = `{"metadata":{"annotations":{"weave.works/kured-node-lock":"'{\"nodeID\":\"manual\"}'"}}}`
+)
 
-func KuredLockExists() (bool, error) {
+func LockExists(client clientset.Interface) (bool, error) {
 	clientSet, err := kubernetes.GetAdminClientSet()
```

Comment:
you can drop this `clientSet` instantiation here, and use the passed `client` below instead